Feat

maksimok93 opened the pull request:
Add patient_search feature

maksimok93 committed:
Feat

Add patient_search feature

maksimok93 committed:
Feat

Add patient_search_step.rb

maksimok93 committed:
Feat

Add auth.feature

maksimok93 committed:
Feat

Add wait_visible_text method

maksimok93 committed:
Docs

Create cucumber.yml for tests reporting in html format.

maksimok93 committed:
Test

Add test scenario Register new patient

maksimok93 committed:
Style

Update step definition in test scenario

maksimok93 committed:
Feat

Add log_in_as_admin method

maksimok93 committed:
Test

Add Rule and Background to Register Patient feature

maksimok93 committed:
Test

Add Background and parameterized Scenario Outline to Patient Search tests

maksimok93 committed:
Exception

Add error exception to step.

maksimok93 committed:
Docs

Update description of test scenarios

maksimok93 committed:
Docs

Update description of test scenarios

maksimok93 committed:
Docs

Add report in html format to the project

AVzayats commented on features/auth.feature:
```diff
@@ -0,0 +1,6 @@
+
+Feature: User authentication functionality
+  Scenario: User with valid credentials can login
+    Given user open main page
```
user opens (no offence, just grammar))

AVzayats commented on features/auth.feature:
```diff
@@ -0,0 +1,6 @@
+Feature: User authentication functionality
+  Scenario: User with valid credentials can login
+    Given user open main page
+    When user logs in with valid username and password
```
better to use other verb than logs in, like 'user enters valid username and password

AVzayats commented on features/patient_search.feature:
```diff
@@ -0,0 +1,19 @@
+      |    'Smith' |        'Smith' |
+      |     'Mary' |         'Mary' |
+
+  Scenario: Find a nonexistent patient
```
Sounds strange, nonexistent patient is in the system)

AVzayats commented on features/step_definitions/register_patient_steps.rb:
```diff
@@ -0,0 +1,53 @@
+  @browser.element(css: 'option[value="M"]').click
+end
+
+And(/^Fill birthdate field$/) do
```
It is better to avoid hard coding and set values as parameters

maksimok93 committed:
Test

Update steps description

maksimok93 committed:
Feat

Initialized module locators.rb; Update css and xpath links in test steps.

maksimok93 committed:
Docs

Add SysAdminLocators, DataManagementLocators

maksimok93 committed:
Test

Add feature System Administration functionality

maksimok93 committed:
Docs

Add reserve OpenMRS link to @browser.goto method.

maksimok93 committed:
Feat

Add screenshot attaching if specific test fails

maksimok93 committed:
Test

Update Scenario Outline: Find an existent patient

maksimok93 committed:
Test

Add tests for next scenario: Merge records for two patients

maksimok93 committed:
Feat

Add exception handling to 'Then' step: page return similar patient found message

maksimok93 committed:
Style

Update DataManagementLocators

maksimok93 committed:
Fix

Delete exception handling in next step: page return record containing patients named 'Daniele'

maksimok93 committed:
Style

Update arguments in set_patients_id method
